lib/core/batchingReceiver.ts

```diff
@@ -206,6 +240,7 @@ export class BatchingReceiver extends MessageReceiver {
         if (reuse) msg += " Receiver link already present, hence reusing it.";
         log.batching(msg, this._context.namespace.connectionId, maxWaitTimeInSeconds, this.name);
         waitTimer = setTimeout(actionAfterWaitTimeout, (maxWaitTimeInSeconds as number) * 1000);
+        resetTimerOnNewMessageReceived();
```

Comment:
If I understand this correctlt, `addCreditAndSetTimer` is called at the very beginning, before any messages are recieved.
Why do we need to set the timer here?
Isnt it enough to set the timer after the first message is recieved?

Otherwise, if the first message itself took longer than `maxMessageWaitTimeoutInSeconds`, then we exit and return an empty array to the user.
The user might be wondering why they got an empty array even when the queue has messages and the `maxWaitTimeInSeconds` has not been reached

Reply:
I see your point. I was thinking about the scenario where there are no active messages in the queue at this moment. Now the link is created and we wait unneccessarily for 60 secs or the specified amount of time.
This ensures that the operation stops asap if it does not receive any messages (first or nth).

What should be the expected behavior?

Reply:
Arent there any apis to get the number of messages in the queue at any given point in time?

Reply:
It's the [http management client API](https://github.com/Azure/azure-service-bus-dotnet/blob/1d5df378a5a2d9622ef3118a8bbf3aed549ed34e/src/Microsoft.Azure.ServiceBus/Management/ManagementClient.cs#L279) that needs to be implemented.

I can add a TODO for now and proceed ahead with waiting after the first message is received.

Reply:
Another point to think about is: If the first message was delayed due to network latency, then making the management api call would most probably have the same fate. Moreover, it makes things a little complex where 
- what if the first message appeared while the http request was in flight
- what if the making the http request took more time, (node.js doesn't have an out of the box way to cancel an issued http request).

Reply:
@amarzavery do you know what the .NET SDK does in that case?

Reply:
spoke to neeraj and he said that in the .net sdk, maxMessageWaitTimeout gets activated after the receiver link is created. I think it is ok to activate the timer after the first message. We can always tweak it later if we see issues.